src/igvfd/types/sample.py

```diff
@@ -200,6 +177,98 @@ def taxa(self, request, donors):
+        if self.item_type == 'primary_cell':
+            if 'cell' not in term_name:
+                summary_terms = f'{term_name} cell'
+            else:
+                summary_terms = term_name
+        elif self.item_type == 'in_vitro_system':
+            summary_terms = f'{term_name} {classification}'
+            if classification in term_name:
+                summary_terms = term_name
+            elif 'cell' in classification and 'cell' in term_name:
+                summary_terms = term_name.replace('cell', classification)
+            elif 'tissue' in classification and 'tissue' in term_name:
+                summary_terms = term_name.replace('tissue', classification)
+        elif self.item_type == 'tissue':
+            if 'tissue' not in term_name:
+                summary_terms = f'{term_name} tissue'
+            else:
+                summary_terms = term_name
+        elif self.item_type == 'whole_organism':
+            summary_terms = concat_numeric_and_units(len(donors), term_name, no_numeric_on_one=True)
+        if cellular_sub_pool and self.item_type in ['primary_cell']:
+            summary_terms = f'{summary_terms} ({cellular_sub_pool})'
+        if embryonic and self.item_type in ['primary_cell', 'tissue']:
+            summary_terms = f'embryonic {summary_terms}'
+        if virtual and self.item_type in ['primary_cell', 'in_vitro_system', 'tissue']:
+            summary_terms = f'virtual {summary_terms}'
+        if time_post_change and time_post_change_units and self.item_type in ['in_vitro_system']:
+            time_post_change = concat_numeric_and_units(time_post_change, time_post_change_units)
+            if targeted_sample_term:
+                targeted_term_object = request.embed(targeted_sample_term, '@@object?skip_calculated=true')
+                targeted_term_name = targeted_term_object.get('term_name')
+                summary_terms = f'{summary_terms} induced to {targeted_term_name} for {time_post_change}'
+            else:
+                summary_terms = f'{summary_terms} induced for {time_post_change}'
+        summary_terms = f'{summary_terms},'
+        if sex and self.item_type in ['primary_cell', 'in_vitro_system', 'tissue', 'whole_organism']:
+            if sex != 'unspecified':
+                if sex == 'mixed':
+                    sex = 'mixed sex'
+                elif len(donors) > 1:
+                    sex = f'{sex}s'
+                summary_terms = f'{summary_terms} {sex}'
+        if taxa and self.item_type in ['primary_cell', 'in_vitro_system', 'tissue', 'whole_organism']:
+            if taxa == 'Mus musculus':
+                strains = []
+                for donor in donors:
+                    donor_object = request.embed(donor, '@@object?skip_calculated=true')
+                    strains.append(donor_object['strain'])
+                strains = ', '.join(sorted(list(set(strains))))
+                taxa = f'{taxa} {strains}'
```

Comment:
i am not clear what happens if there is mix of taxas?

Reply:
No strains are listed if taxa is mixed.

Reply:
I can add strains if mixed taxa includes mice, I excluded since I assumed taxa would be excluded if it were mixed.